src/main/java/com/bobocode/hoverla/bring/BringApplication.java

```diff
@@ -9,18 +19,113 @@
+ *              .build();
+ *      String bean = context.getBean("BeanName");
+ * }
+ *
+ * It is necessary to provide packages will be scan to define bean definition configs. User can pass either
+ * package names or user class that is in package which should be scanned.
  */
 public class BringApplication {
+    private BringApplication() {
+    }
+
     /**
      * Initializes and returns {@link ApplicationContext}
+     * @param packagesToScan list of packages to scan
+     */
+    public static ApplicationContext loadContext(String... packagesToScan) {
+        return getContext(packagesToScan);
+    }
+
+    /**
+     * Initializes and returns {@link ApplicationContext}
+     * @param clazz particular user class that is in the package which is candidate for scanning
+     */
+    public static ApplicationContext loadContext(Class<?> clazz) {
+        return getContext(clazz.getPackageName());
+    }
+
+    /**
+     * Initializes input parameters (list of {@link BeanScanner}, {@link BeanDefinitionValidator}, {@link BeanInitializer})
+     * for ApplicationContext creation and return instance of context
+     *
+     * @param packagesToScan list of packages for scanning to define bean configs
+     * @return instance of {@link ApplicationContextImpl}
+     */
+    private static ApplicationContext getContext(String... packagesToScan)  {
+        if (packagesToScan == null || packagesToScan.length == 0) {
+            throw new IllegalArgumentException("Argument [packagesToScan] must contain at least one element");
+        }
+        // TODO: add initialization of scanners;
+        List<BeanScanner> scanners = List.of();
+
+        return new ApplicationContextImpl(scanners, new BeanDefinitionValidator(), new BeanInitializer());
+    }
+
+
+    /**
+     * Creates instance of ApplicationContextBuilder class
+     * @return ApplicationContextBuilder instance
+     */
+    static ApplicationContextBuilder getContextBuilder() {
+        return new ApplicationContextBuilder();
+    }
+
+    /**
+     *
      */
-    public static ApplicationContext loadContext() {
-        return null;
+    static class ApplicationContextBuilder {
+        private Level logLevel;
+        private String[] packagesToScan;
+        private Class<?> classToScan;
+
+        public ApplicationContextBuilder logLevel(Level logLevel) {
+            this.logLevel = logLevel;
+            return this;
+        }
+
+        public ApplicationContextBuilder packagesToScan(String... packagesToScan) {
+            this.packagesToScan = packagesToScan;
+            return this;
+        }
+
+        public ApplicationContextBuilder classToScan(Class<?> classToScan){
+            this.classToScan = classToScan;
+            return this;
+        }
+
+        public ApplicationContext build() {
+            Logger logger = (Logger) LoggerFactory.getLogger(org.slf4j.Logger.ROOT_LOGGER_NAME);
+            logger.setLevel(logLevel != null ? logLevel : Level.INFO);
```

Comment:
I'd like you to record a video or show on a call that log level really changes without intruding into log logback xml. I mean I'd like to see it end-to-end